### PR TITLE
Add SET NULL / SET DEFAULT examples to FK doc

### LINKED ISSUE #2486: Add SET NULL and SET DEFAULT Examples to Foreign Key
re: #2015, #2379

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2486.

### CODE REVIEW

File: v2.0/foreign-key.md [hunk 1]
```diff
@@ -343,6 +343,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE orders_3 (
+    id INT PRIMARY KEY,
+    customer_id INT REFERENCES customers_3(id) ON UPDATE SET NULL ON DELETE SET NULL
+  );
+~~~
+
+Insert a few records in the referenced table:
```
Comment (jseldess) [inline v2.0/foreign-key.md]: nit: `in the referenced table` > `into the referenced table`

Same for the `CASCADE` example.

File: v2.0/foreign-key.md [hunk 2]
```diff
@@ -343,6 +343,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
+
+{% include copy-clipboard.html %}
+~~~ sql
+> INSERT INTO customers_3 VALUES (1), (2), (3);
+~~~
+
+Insert some records in the child table:
```
Comment (jseldess) [inline v2.0/foreign-key.md]: Same as above.

File: v2.0/foreign-key.md [hunk 3]
```diff
@@ -343,6 +343,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
+| id  | customer_id |
++-----+-------------+
+| 100 |        NULL |
+| 101 |           2 |
+| 102 |           3 |
+| 103 |        NULL |
++-----+-------------+
+~~~
+
+When the `id = 1` was updated to `id = 23` in `customers_3`, the referencing `customer_id` was set to `NULL`.
```
Comment (jseldess) [inline v2.0/foreign-key.md]: nit: `When the id = 1` > `When id = 1`

Same for other examples.

File: v2.0/foreign-key.md [hunk 4]
```diff
@@ -343,6 +343,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
+| id  | customer_id |
++-----+-------------+
+| 100 |        9999 |
+| 101 |           2 |
+| 102 |           3 |
+| 103 |        9999 |
++-----+-------------+
+~~~
+
+When the `id = 1` was updated to `id = 23` in `customers_4`, the referencing `customer_id` was set to `DEFAULT` (i.e., `9999`). You can see this in the first row of `orders_4`, where `id = 100` and the `customer_id` is now `9999`
```
Comment (jseldess) [inline v2.0/foreign-key.md]: nit: `You can see this in the first row` > `You can see this in the first and last rows`

File: v2.0/foreign-key.md [hunk 5]
```diff
@@ -343,6 +343,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
++------+
+|   id |
++------+
+|    3 |
+|   23 |
+| 9999 |
++------+
+~~~
+
+Let's check to make sure the row in `orders_4` where `customers_id = 2` was updated to the `DEFAULT` value (i.e., `9999`):
```
Comment (jseldess) [inline v2.0/foreign-key.md]: `make sure the row` > `make sure the rows`

File: v2.0/foreign-key.md (hunk 1 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: Done

File: v2.0/foreign-key.md (hunk 2 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: Done

File: v2.0/foreign-key.md (hunk 3 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: Done

File: v2.0/foreign-key.md (hunk 4 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: Done

File: v2.0/foreign-key.md (hunk 5 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: This one is just one row. I just deleted `id = 2` so the `orders_4.id = 101` row was set to default. I edited the language to hopefully clarify. 

Was just thinking - can I add a visual marker (like a `<`) to the response? That might help clarify things. Something like this:

~~~
+-----+-------------+
| id      |  customer_id |
+-----+-------------+
| 100   |            9999 |
| 101   |             9999 | <
| 102   |                   3 |
| 103   |            9999 |
+-----+-------------+
~~~

Comment (jseldess) [reply v2.0/foreign-key.md]: Oh, sorry, my mistake. No need to do anything then, I think.

**Comment by jseldess** [comment]
@BramGruneir, I think this looks good to go, but can you also take a look so Lauren can get this merged?

File: v2.0/foreign-key.md [hunk 6]
```diff
@@ -343,6 +349,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
++----+
+|  2 |
+|  3 |
+| 23 |
++----+
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SELECT * FROM orders_3;
```
Comment (BramGruneir) [inline v2.0/foreign-key.md]: It might be nice to show the before and after for these.

File: v2.0/foreign-key.md [hunk 7]
```diff
@@ -343,6 +349,232 @@ Let's check to make sure the rows in `orders_2` where `customers_id = 23` were a
...
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE orders_4 (
+    id INT PRIMARY KEY,
+    customer_id INT DEFAULT 9999 REFERENCES customers_4(id) ON UPDATE SET DEFAULT ON DELETE SET DEFAULT
+  );
+~~~
+
+Insert a few records into the referenced table:
```
Comment (BramGruneir) [inline v2.0/foreign-key.md]: Might be a good idea to call out the 9999 since it will be used as the default.

File: v2.0/foreign-key.md (hunk 6 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: Done

File: v2.0/foreign-key.md (hunk 7 above)
Comment (lhirata) [reply v2.0/foreign-key.md]: Done

**Comment by lhirata** [comment]
@BramGruneir thanks for the review!

**Pull request merged**